app/services/RestaurantService.java

```diff
@@ -102,14 +111,28 @@ public Boolean deleteRestaurant(final UUID id) throws Exception {
 
 		criteria.addOrder(Order.asc("name"));
 
+		criteria.setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY);
+		
 		List<Restaurant> restaurants = criteria.list();
+		List<Restaurant> tmpList = new ArrayList<>();
+
+		if (restaurantFilter.rating != null && restaurantFilter.rating > 0) {
+			for (Restaurant restaurant: restaurants)
+			{
+				if(restaurant.getAverageRating() >= restaurantFilter.rating-0.5 && restaurant.getAverageRating() < restaurantFilter.rating+0.5)
+					tmpList.add(restaurant);
+			}
+			restaurants = tmpList;
+			numberOfPages =  new Long(restaurants.size()) / restaurantFilter.pageSize;
```

Comment:
Will this not change the number of restaurants per page? Say we fetched 10 restaurants on line 116 from the Database, if we remove them after that, the list will have less than (and possibly even 0) restaurants on the page. It will also change from page to page. All result filtering must be done in the database query.